### PR TITLE
Implemented support to store user preference for selected product store(#2f2h90a)

### PR DESCRIPTION
**Pull request by rathoreprashant**
### Related Issues
Added user preference for selected product store

Closes #

 ### Short Description and Why It's Useful
 

 ### Screenshots of Visual Changes before/after (If There Are Any)
 

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [ ] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/services/UserService.ts [hunk 1]
```diff
@@ -40,10 +40,28 @@ const getEComStores = async (payload: any): Promise<any> => {
   });
 }
 
+const setUserPref = async (payload: any): Promise<any> => {
```
Comment (adityasharma7) [inline src/services/UserService.ts:43]:
```suggestion
const setUserPreference = async (payload: any): Promise<any> => {
```

File: src/store/modules/user/actions.ts [hunk 2]
```diff
@@ -77,9 +77,13 @@ const actions: ActionTree<UserState, RootState> = {
 
         const eComStoreResp = await UserService.getEComStores(payload);
         if (eComStoreResp.status === 200 && eComStoreResp.data.docs?.length > 0 && !hasError(eComStoreResp)) {
+          const userPref =  await UserService.getUserPref({
+            'userPrefTypeId': 'SELECTED_BRAND'
+          });
           const stores = eComStoreResp.data.docs
+          const store = stores.find((store: any) => store.productStoreId == userPref.data.userPrefValue)
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:84]: Improve variable name `store`. It should based upon what value we are storing in it

File: src/store/modules/user/actions.ts [hunk 3]
```diff
@@ -77,9 +77,13 @@ const actions: ActionTree<UserState, RootState> = {
...
           resp.data.stores = stores ? stores : [];
-          commit(types.USER_CURRENT_ECOM_STORE_UPDATED, stores ? stores[0] : {});
+          commit(types.USER_CURRENT_ECOM_STORE_UPDATED, stores ? (store ? store : stores[0]) : {});
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:86]: We should check if store has value, if not set the first store from stores list

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/services/UserService.ts [hunk 4]
```diff
@@ -40,10 +40,28 @@ const getEComStores = async (payload: any): Promise<any> => {
...
+  return api({
+    url: "service/setUserPreference",
+    method: "post",
+    data: payload
+  });
+}
+
+const getUserPref = async (payload: any): Promise<any> => {
```
[the remark this answers is not in the document]
Comment (rathoreprashant) [reply src/services/UserService.ts]: Changes Applied sir

File: src/store/modules/user/actions.ts (hunk 2 above)
Comment (rathoreprashant) [reply src/store/modules/user/actions.ts]: Improved sir!

File: src/services/UserService.ts [hunk 5]
```diff
@@ -40,10 +40,28 @@ const getEComStores = async (payload: any): Promise<any> => {
...
 
+const setUserPreference = async (payload: any): Promise<any> => {
+  return api({
+    url: "service/setUserPreference",
+    method: "post",
+    data: payload
+  });
+}
+
+const getUserPreference = async (payload: any): Promise<any> => {
```
Comment (adityasharma7) [inline src/services/UserService.ts:51]: Add a TODO note here for why we are using post method for get operation

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/services/UserService.ts (hunk 5 above)
Comment (rathoreprashant) [reply src/services/UserService.ts:51]: comment Added sir

File: src/services/UserService.ts [hunk 6]
```diff
@@ -40,10 +40,29 @@ const getEComStores = async (payload: any): Promise<any> => {
...
+  return api({
+    url: "service/getUserPreference",
+    //TODO Due to security reasons service model OMS 1.0 does not support sending parameters in get request that's why we use post here
```
Comment (adityasharma7) [inline src/services/UserService.ts:54]:
```suggestion
    //TODO Due to security reasons service model of OMS 1.0 does not support sending parameters in get request that's why we use post here
```

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**